tests/test_remaining_graph_rules.py

```
@@ -2476,3 +2484,374 @@ def test_ignores_data_flow_descendants_outside_play(self, rule: ListAllUsedVaria
+        assert result is not None
+        assert result.detail is not None
+        var_list = cast(list[dict[str, object]], result.detail["variable_set"])
+        assert any(v["name"] == REDACTED for v in var_list)
+
+    def test_misnamed_scalar_secret_redacted(self, rule: ShowVariablesGraphRule) -> None:
+        """Cleartext credentials in generically named vars are detected and redacted.
+
+        Args:
+            rule: Rule instance under test (unused; kept for fixture symmetry).
+        """
+        secret = "mysql://user:secret@db.example.com/prod"
```

Ruff (0.16.2)

[error] Possible hardcoded password assigned to: "secret"

(S105)